### PR TITLE
Admin user page

### LINKED ISSUE #292: Implement User Role Management Page
As we expand the roles available in Bitcoin Transcripts beyond just 'reviewers', there is a need for a way to manage these roles effectively. To facilitate this, we will create a new admin page (`/admin/users`) dedicated to user and roles management.

**Features of the User Management Page**
- **User Data Display**: Utilize the `api/users` API endpoint to fetch and display a list of all users. The table will include user data such as id, GitHub username, role, and creation date. This straightforward presentation will help admins quickly understand the current user base and their roles.
- **Role Adjustment Capability**: Admins will have the ability to select a user from the table and change their role using the `/api/users/{id}` API endpoint.
- **No Filtering**: Initially, there will be no need to implement filtering or advanced search functionalities as seen in other admin pages.

**Access Control**
This page will be exclusively accessible to users with admin privileges.

### PR DESCRIPTION
**Pull request by 0tuedon**
This PR closes #292

### CODE REVIEW

**Comment by kouloumos** [comment]
@0tuedon Where is the new page? I run the branch locally but I can't see anything new.

**Comment by 0tuedon** [comment]
> @0tuedon Where is the new page? I run the branch locally but I can't see anything new.

Added the route on the navbar now You should be able to see it now

**Comment by kouloumos** [comment]
image

This doesn't seem right. You are showing duplicate information. Once in the "Role Permissions" columns (btw just call it "Role") and again on the dropdown. 

I don't like the dropdown for all users. Can we try to use a checkbox that when clicked will display a button alongside this dropdown button on the top (same as in the other tables).

Page should be called "Users", same for the route (I explicitly mentioned it on the issue)

**Review by kouloumos — COMMENTED** [review]
Looks good! :rocket: 

The only observation that I have is the usage of "AllUsers" in the variable names. You mentioned the possibility of adding filters to this table, so when that happens, "allusersX" is not gonna be relevant and the diff will need to be bigger and extend into multiple files.

**Comment by 0tuedon** [comment]
> X

Kind of, but it also depends on how many filters we are handling, for example, other pages with the `BaseTable` have the same format.

File: src/config/ui-config.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export const ROUTES_CONFIG = {
   TRANSACTIONS: "admin/transactions",
   ALL_REVIEWS: "admin/reviews?status=active",
   REVIEWS: "reviews",
+  ALL_USERS: "admin/users",
```
Comment (kouloumos) [inline src/config/ui-config.ts:12]: you missed this one sir

File: src/services/api/admin/useUsers.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  return axios
+    .get(endpoints.GET_USERS_ADMIN())
+    .then((res) => res.data)
+    .catch((err) => err);
+};
+
+export const useGetUsers = () =>
+  useQuery({
+    queryFn: () => getUsers(),
+    queryKey: ["all_users"],
```
Comment (kouloumos) [inline src/services/api/admin/useUsers.ts:25]: you missed this one sir

**Review by kouloumos — COMMENTED** [review]
minor changes for completeness after the "allusersX" -> "usersX" change :eye:

File: src/config/ui-config.ts (hunk 1 above)
Comment (0tuedon) [reply src/config/ui-config.ts:12]: Oh, darnit

**Review by kouloumos — APPROVED** [review]
:rocket:

**Pull request merged**